### PR TITLE
Add a bunch of actions for the webapp

### PR DESCRIPTION
**Pull request by jwilander**
#### Summary
Add a bunch of actions/selectors for the webapp and do some clean-up.

#### Checklist
- [x] Added or updated unit tests (required for all new features)
- [x] All new/modified APIs include changes to the drivers

### CODE REVIEW

**Comment by jwilander** [comment]
@enahum For changes you might need to care about for RN:
* Removed `updateNotifyProps` action and added `updateMe` which just does a patch (so just pass it `{notify_props: propsgohere}`)
 * I also didn't make it optimistic because I don't think we want it to be
* Refactored the `userAutocompleteInChannel` to be general `userAutocomplete`
* Switched the selector names for users from `getProfilesInThing` to `getUserIdsInThings` so it's more clear what they do

**Comment by jwilander** [comment]
@jarredwitt in case you want to take a look as well

**Comment by enahum** [comment]
@jarredwitt there are changes in this PR that we need to consider for RN, I know I've done a bunch of changes for https://github.com/mattermost/mattermost-mobile/pull/484 so lets merge that one first and then we can make the necessary changes if any

File: src/actions/users.js [hunk 1]
```diff
@@ -339,7 +408,7 @@ export function getProfilesInChannel(channelId, page, perPage = General.PROFILE_
             },
             {
                 type: UserTypes.RECEIVED_PROFILES_LIST,
-                data: profiles
+                data: removeUserFromList(currentUserId, Object.assign([], profiles))
```
Comment (enahum) [inline src/actions/users.js]: I think here instead of Object.assign you can do `[...profiles]`

File: src/actions/users.js [hunk 2]
```diff
@@ -514,41 +591,62 @@ export function getUserAudits(userId, page = 0, perPage = General.AUDITS_CHUNK_S
...
+                    id: channelId
+                }
+            );
+        }
+
+        if (teamId) {
+            actions.push(
+                {
+                    type: UserTypes.RECEIVED_PROFILES_LIST_IN_TEAM,
+                    data: Object.assign([], data.users, data.out_of_channel),
```
Comment (enahum) [inline src/actions/users.js]: same here, and a few more above and below

**Comment by jwilander** [comment]
@enahum changes made

**Pull request merged**